yt_dlp/extractor/dropout.py

```diff
@@ -190,29 +194,44 @@ class DropoutSeasonIE(InfoExtractor):
                 'id': 'dimension-20-shriek-week-season-1',
                 'title': 'Dimension 20 Shriek Week - Season 1'
             }
+        },
+        {
+            'url': 'https://www.dropout.tv/breaking-news-no-laugh-newsroom/season:3',
+            'note': 'Multi-season series with season in the url that requires pagination',
+            'playlist_count': 25,
+            'info_dict': {
+                'id': 'breaking-news-no-laugh-newsroom-season-3',
+                'title': 'Breaking News No Laugh Newsroom - Season 3'
+            }
         }
     ]
 
+    def _fetch_page(self, url, season_id, page):
+        page += 1
+        try:
+            webpage = self._download_webpage(
+                f'{url}?page={page}',
+                season_id,
+                note=f'Downloading page {page}'
+            )
+            for item in get_elements_by_class('js-collection-item', webpage):
+                yield self.url_result(
+                    url=self._search_regex(r'<a href=["\'](.+?)["\'] class=["\']browse-item-link["\']',
+                                           item, 'item_url'),
+                    ie=DropoutIE.ie_key()
+                )
+        except Exception as e:
+            if e.exc_info[0] == HTTPError and e.exc_info[1].code == 400:  # Site returns 400 when you page past the end
+                return
+            else:
+                raise
+
     def _real_extract(self, url):
         season_id = self._match_id(url)
+        season_num = self._match_valid_url(url).group('season') or 1
         season_title = season_id.replace('-', ' ').title()
-        webpage = self._download_webpage(url, season_id)
-
-        entries = [
-            self.url_result(
-                url=self._search_regex(r'<a href=["\'](.+?)["\'] class=["\']browse-item-link["\']',
-                                       item, 'item_url'),
-                ie=DropoutIE.ie_key()
-            ) for item in get_elements_by_class('js-collection-item', webpage)
-        ]
 
-        seasons = (get_element_by_class('select-dropdown-wrapper', webpage) or '').strip().replace('\n', '')
-        current_season = self._search_regex(r'<option[^>]+selected>([^<]+)</option>',
-                                            seasons, 'current_season', default='').strip()
+        entries = OnDemandPagedList(functools.partial(
+            self._fetch_page, url, season_id), self._PAGE_SIZE)
 
-        return {
-            '_type': 'playlist',
-            'id': join_nonempty(season_id, current_season.lower().replace(' ', '-')),
-            'title': join_nonempty(season_title, current_season, delim=' - '),
-            'entries': entries
-        }
+        return self.playlist_result(entries, playlist_id=f'{season_id}-season-{season_num}', playlist_title=f'{season_title} - Season {season_num}')
```

Comment:
line too long
```suggestion
        return self.playlist_result(
            entries, f'{season_id}-season-{season_num}', f'{season_title} - Season {season_num}')
```

&nbsp;
or better imo, you could inline entries (and remove var above)
```suggestion
        return self.playlist_result(
            OnDemandPagedList(functools.partial(self._fetch_page, url, season_id), self._PAGE_SIZE),
            f'{season_id}-season-{season_num}', f'{season_title} - Season {season_num}')
```